nilearn/glm/thresholding.py

```diff
@@ -412,6 +395,38 @@ def threshold_stats_img(
         without correction.
 
     """
+    if threshold is not None and height_control is not None:
+        # TODO (nilearn >= 0.15.0) update 3.09 to DEFAULT_Z_THRESHOLD
+        # if user specifies threshold
+        if float(threshold) != 3.0:
+            warnings.warn(
+                f"\n'{threshold=}' is not used with '{height_control=}'."
+                "\n'threshold' is only used when 'height_control=None'. "
+                "\nEither set 'height_control=None' or 'threshold=None' to "
+                "avoid this warning.",
+                UserWarning,
+                stacklevel=find_stack_level(),
+            )
+            height_control = None
```

Comment:
actually I don't think we should set `height_control=None`, I prefer to have more friction if users want to NOT do any multiple test correction.